packages/lisk-transactions/src/transactions/3_vote_transaction.ts

```diff
@@ -0,0 +1,459 @@
+		voteTransaction.sign(passphrase, secondPassphrase);
+
+		return voteTransaction.toJSON();
+	}
+
+	public static fromJSON(tx: TransactionJSON): VoteTransaction {
+		const transaction = new VoteTransaction(tx);
+		const { errors, status } = transaction.validateSchema();
+
+		if (status === Status.FAIL && errors.length !== 0) {
+			throw new TransactionMultiError(
+				'Failed to validate schema.',
+				tx.id,
+				errors,
+			);
+		}
+
+		return transaction;
+	}
+
+	protected getAssetBytes(): Buffer {
+		return Buffer.from(this.asset.votes.join(''), 'utf8');
+	}
+
+	public assetToJSON(): object {
+		return {
+			votes: this.asset.votes,
+		};
+	}
+
+	public getRequiredAttributes(): Attributes {
+		const attr = super.getRequiredAttributes();
+		const publicKey = this.asset.votes.map(pkWithAction =>
+			pkWithAction.slice(1),
+		);
+
+		return {
+			[ENTITY_ACCOUNT]: {
+				...attr[ENTITY_ACCOUNT],
+				publicKey,
+			},
+		};
+	}
+
+	public verifyAgainstOtherTransactions(
+		transactions: ReadonlyArray<TransactionJSON>,
+	): TransactionResponse {
+		const sameTypeTransactions = transactions
+			.filter(
+				tx =>
+					tx.senderPublicKey === this.senderPublicKey && tx.type === this.type,
+			)
+			.map(tx => new VoteTransaction(tx));
+		const publicKeys = this.asset.votes.map(vote => vote.substring(1));
+
+		const errors = sameTypeTransactions.reduce(
+			(previous, tx) => {
+				if (
+					tx.asset.votes
+						.map(vote => vote.substring(1))
+						.some(publicKey => publicKeys.includes(publicKey))
+				) {
+					return [
+						...previous,
+						new TransactionError(
+							'Transaction includes conflicting vote',
+							tx.id,
+							'.asset.votes',
+						),
+					];
+				}
+
+				return previous;
+			},
+			[] as ReadonlyArray<TransactionError>,
+		);
+
+		return {
+			id: this.id,
+			errors,
+			status: errors.length === 0 ? Status.OK : Status.FAIL,
+		};
+	}
+
+	public processRequiredState(state: EntityMap): RequiredVoteState {
+		const accounts = state[ENTITY_ACCOUNT];
+		if (!accounts) {
+			throw new Error('Entity account is required.');
+		}
+		if (
+			!isTypedObjectArrayWithKeys<Account>(accounts, ['address', 'publicKey'])
+		) {
+			throw new Error('Required state does not have valid account type.');
+		}
+
+		const sender = accounts.find(acct => acct.address === this.senderId);
+		if (!sender) {
+			throw new Error('No sender account is found.');
+		}
+		const votes = this.asset.votes.map(vote => vote.substring(1));
+		const dependentAccounts = accounts.filter(acct =>
+			votes.includes(acct.publicKey),
+		);
+		if (votes.length !== dependentAccounts.length) {
+			throw new Error('Not enought accounts in dependent state.');
+		}
+
+		return {
+			sender,
+			dependentState: {
+				[ENTITY_ACCOUNT]: dependentAccounts,
+			},
+		};
+	}
+
+	public validateSchema(): TransactionResponse {
+		const { errors: baseErrors, status } = super.validateSchema();
+		const valid = validator.validate(voteAssetFormatSchema, this.asset);
```

Comment:
validate whole schema format here?

Reply:
basic schema is validated at `super.validateSchema` so I think validating the asset only is good here